system_modules/napmath/src/mathutils.cpp

```diff
@@ -27,7 +27,7 @@ namespace nap
 
 		static std::mt19937& getGenerator()
 		{
-			static std::unique_ptr<std::mt19937> generator = nullptr;
+			static thread_local std::unique_ptr<std::mt19937> generator = nullptr;
```

Comment:
I guess this is ok, assuming we want different random sequences for every thread. I can't think of a situation where this would be a problem atm, but considering the current implementation is not thread safe this solutions outweighs the other.

Reply:
Interestingly, declaring something `static` in the cpp (such as the generator) is thread safe since C++ 11. Also, looking at the implementation it can be simplified to:

```
		static std::mt19937& getGenerator()
		{
			static thread_local std::mt19937 generator(time(0));
			return generator;
		}

```

I'll implement and test that some other time.